test/helper.sh

```diff
@@ -0,0 +1,216 @@
+#!/bin/bash
+
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+ZONE="us-west1-a"
+SUBNET_REGION="us-west1"
+
+###############################################################################
+# Genereate a hash of length 20 using sha1 checksum and take the first 20 characters.
+# Globals:
+#   None
+# Argument:
+#   Value to be hashed, a string
+# Outputs:
+#   Writes the hashed result to stdout.
+###############################################################################
+get_hash() {
+    # By default, sha1sum prints out hash and filename, so we only access the 
+    # [0] element for the hash.
+    h=($(echo -n $1 | sha1sum))
+    echo "${h:0:20}"
+}
+
+###############################################################################
+# Wait for the given ingress to be fully provisioned.
+# Globals:
+#   None
+# Argument:
+#   Name of the ingress.
+#   Namespace of the ingress.
+# Outputs:
+#   Writes the ingress IP to stdout.
+# Returns:
+#   0 if the ingress IP is populated within 15 minutes, 1 if not.
+###############################################################################
+wait_for_ingress_ip() {
+    local ing ns
+    ing="$1"
+    ns="$2"
+
+    for ATTEMPT in $(seq 30); do
+        vip=$(kubectl get ingress ${ing} -n ${ns} -o jsonpath="{.status.loadBalancer.ingress[0].ip}")
+        if [[ ! -z "$vip" ]]; then
+            echo "${vip}"
+            return
+        fi
+        sleep 30
+    done
+    exit 1
+}
+
+###############################################################################
+# Wait for the given Managed Certificate to be fully provisioned.
+# Globals:
+#   None
+# Arguments:
+#   Name of the Managed Certificate.
+#   Namespace of the Managed Certificate.
+# Returns:
+#   0 if the status of the given certificate is Active within 45 minutes, 1 if not.
+###############################################################################
+wait_for_managed_cert() {
+    local mc ns
+    mc="$1"
+    ns="$2"
+
+    for ATTEMPT in $(seq 45); do
```

Comment:
45 *60s is 45 minutes.

Do we need to wait this long? I think at most this should wait 5 - 10 minutes. You should also look at the default go test timeout, as this probably exceeds that.